ouroboros-consensus/src/Ouroboros/Consensus/Util.hs

```diff
@@ -381,6 +405,17 @@ allDisjoint = go Set.empty
 
 (......:) :: (y -> z) -> (x0 -> x1 -> x2 -> x3 -> x4 -> x5 -> x6 -> y) -> (x0 -> x1 -> x2 -> x3 -> x4 -> x5 -> x6 -> z)
 (f ......: g) x0 x1 x2 x3 x4 x5 x6 = f (g x0 x1 x2 x3 x4 x5 x6)
+
+{-------------------------------------------------------------------------------
+  Type-level composition
+-------------------------------------------------------------------------------}
+
+type (:..:) :: (m -> Type) -> (k -> l -> m) -> k -> l -> Type
```

Comment:
I wonder if we could add here some examples to form an intuition about this operator. Eg. examples involving `Either` and `Identity`, or `Maybe`.